corehq/apps/app_manager/xpath.py

```diff
@@ -40,8 +40,12 @@ def interpolate_xpath(string, case_xpath=None, fixture_xpath=None, module=None,
     """
     Replace xpath shortcuts with full value.
     """
-    from corehq.apps.app_manager.util import xpath_references_case
-    if case_xpath is None and xpath_references_case(string):
+    if case_xpath is None and any([
+        '#case' in string,
+        '#parent' in string,
+        '#host' in string,
+        re.search(DOT_INTERPOLATE_PATTERN, string),
+    ]):
```

Comment:
this reverts this function to validating only what it needs to ensure that it can use `case_xpath` if the `string` requires it.

with this change, module filters go back to what they were before https://github.com/dimagi/commcare-hq/pull/10972, and form filters stay the same as in https://github.com/dimagi/commcare-hq/pull/10972 since validation for those happens elsewhere